### ISSUE TITLE
Register tracers in dispatcher via method instead of constructor

### ISSUE THREAD

**Issue by cpx86** [openzipkin/zipkin4net#210]
Just realized I missed a very important detail with #208.

The way the TraceManager handles the default dispatcher, it registers it's internal Push method via the constructor of the dispatcher (see https://github.com/openzipkin/zipkin4net/blob/master/Src/zipkin4net/Src/TraceManager.cs#L61). This is obviously not possible for any dispatcher defined outside of zipkin4net.

One way to solve this would be to extend the IRecordDispatcher interface so that the tracer action is registered via a method instead of the constructor. Either
1. Add the `Action<Record>` parameter to the existing `Record` method
2. Add a new method e.g. `RegisterTracers`

Thoughts?

**Comment by fedj** [comment]
Hi @cpx86,

I'm not sure to understand. Can't you use Start(ILogger, IRecordDispatcher) ?

**Comment by fedj** [comment]
Oh, just saw the problem, let me check

**Comment by fedj** [comment]
It weirdly sound to me that tracers should be registered in dispatchers. I would vote for a RegisterTracer method in Dispatcher.

**Comment by cpx86** [comment]
I think that makes sense as well.

Another question - should the dispatcher receive an pre-baked `Action<Record>` or would it make sense to just send an `IEnumerable<Tracer>` instead? I'm leaning towards the latter - feels more clean.

**Comment by fedj** [comment]
Indeed, feels cleaner as it would separate concerns. Can you come up with a PR ?
